test/e2eERC20.js

```diff
@@ -0,0 +1,116 @@
+    const chainID = 1;
+
+    const depositerAddress = accounts[1];
+    const recipientAddress = accounts[2];
+    const relayer1Address = accounts[3];
+    const relayer2Address = accounts[4];
+
+    const initialTokenAmount = 100;
+    const depositAmount = 10;
+    const expectedDepositNonce = 1;
+
+    let RelayerInstance;
+    let BridgeInstance;
+    let ERC20MintableInstance;
+    let ERC20HandlerInstance;
+
+    let tokenID;
+    let depositData;
+    let depositProposalData;
+    let depositProposalDataHash;
+
+    beforeEach(async () => {
+        RelayerInstance = await RelayerContract.new([relayer1Address, relayer2Address], relayerThreshold);
+        BridgeInstance = await BridgeContract.new(chainID, RelayerInstance.address, relayerThreshold);
+        ERC20MintableInstance = await ERC20MintableContract.new();
+        ERC20HandlerInstance = await ERC20HandlerContract.new(BridgeInstance.address);
+
+        await ERC20MintableInstance.mint(depositerAddress, initialTokenAmount);
+        await ERC20MintableInstance.approve(ERC20HandlerInstance.address, depositAmount, { from: depositerAddress });
+
+        await ERC20MintableInstance.addMinter(ERC20HandlerInstance.address);
+
+        tokenID = Ethers.utils.hexZeroPad(Ethers.utils.hexlify(chainID), 32).substr(2) + 
+                  Ethers.utils.hexZeroPad(Ethers.utils.hexlify(ERC20MintableInstance.address), 32).substr(2);
+
+        depositData = '0x' +
+            Ethers.utils.hexZeroPad(ERC20MintableInstance.address, 32).substr(2) +
+            Ethers.utils.hexZeroPad(recipientAddress, 32).substr(2) +
+            Ethers.utils.hexZeroPad(Ethers.utils.hexlify(depositAmount), 32).substr(2);
+
+        depositProposalData = '0x' +
+            Ethers.utils.hexZeroPad(recipientAddress, 32).substr(2) +
+            Ethers.utils.hexZeroPad(Ethers.utils.hexlify(depositAmount), 32).substr(2) +
+            Ethers.utils.hexZeroPad(Ethers.utils.hexlify(64), 32).substr(2) + // length of next arg in bytes
+            tokenID;
+        depositProposalDataHash = Ethers.utils.keccak256(depositProposalData);
+    });
+
+    it("[sanity] depositerAddress' balance should be equal to initialTokenAmount", async () => {
+        const depositerBalance = await ERC20MintableInstance.balanceOf(depositerAddress);
+        assert.strictEqual(depositerBalance.toNumber(), initialTokenAmount);
+    });
+
+    it("[sanity] ERC20HandlerInstance.address should have an allowance of depositAmount from depositerAddress", async () => {
+        const originChainHandlerAllowance = await ERC20MintableInstance.allowance(depositerAddress, ERC20HandlerInstance.address);
+        assert.strictEqual(originChainHandlerAllowance.toNumber(), depositAmount);
+    });
+
+    it("depositAmount of Destination ERC20 should be minted for recipientAddress", async () => {
+        // depositerAddress makes initial deposit of depositAmount
+        TruffleAssert.passes(await BridgeInstance.deposit(
+            chainID,
```

Comment:
although using `chainID` works, its very misleading. you set the bridge contract to this `chainID` and then you also use it as the deposit. It's like making a deposit for the same chain.

I would instead, rename `chainID` to `originChainID` and I would set the bridge contract to `destinationChainID` since the bridge instance you are using is technically the `desitnationID`

Reply:
The test is testing a deposit on the same chain, I do need to change the language in the comments though

Reply:
I was trying to hit [this](https://github.com/ChainSafe/chainbridge-solidity/blob/80b2d8f242aa194d9c4a3fb7a97307038e65cecf/contracts/handlers/ERC20Handler.sol#L140) part of  `executeDeposit`